### DISCUSSION TITLE
Allowing user to update their Name

### DISCUSSION THREAD

**Discussion by slammer1870** [nextauthjs/next-auth#2422] [Help] [answered]
I am using the Email provider for passwordless authentication and for my "newUser" page I bring them to a signup form to allow them to input their name.

My current configuration is to make a post request to a custom api route that updates the user's name in a mongodb database using MongoClient. My issue is figuring out how to update this in the session at the same time.

If a user logs out and logs back in their name will appear in their account page on the frontend but I can't quite figure out how to update it without have to log out and log back in to a new session.

Any feedback would be greatly appreciated!

**Comment by mahieyin-rahmun** [comment, marked as answer]
My approach in this case usually has been to keep only the `userId` in the session object and nothing else, since that will not change. 

Then, I have a `/me` endpoint which ~the~ requires authentication, and that way, any information related to the user can be queried after the user has logged in. I usually store this in a context object which can be accessed anywhere down the component tree.

**Reply by slammer1870** [reply]
@mahieyin-rahmun thank you for your response, that seems like an elegant and simple solution to me, exactly what I was looking for.

P.S. your article on how to configure social authentication with Django has saved me hours of trying to figure this out so thank you so much for the great content!

**Reply by mahieyin-rahmun** [reply]
Glad that I was able to help! Have a good day!
